std/typecons.d

```diff
@@ -80,40 +79,69 @@ public:
     ---
     */
     static Unique!T create(A...)(auto ref A args)
-    if (__traits(compiles, new T(args)))
     {
+        import core.memory : GC;
+        import core.stdc.stdlib : malloc;
+        import std.conv : emplace;
+        import std.exception : enforce;
+
         debug(Unique) writeln("Unique.create for ", T.stringof);
         Unique!T u;
-        u._p = new T(args);
+
+        // TODO: May need to fix alignment?
+        // Does emplace still need to mess with alignment if
+        // the memory is coming from malloc, or does malloc handle that?
+
+        static if (is(T == class))
+            immutable size_t allocSize = __traits(classInstanceSize, T);
+        else
+            immutable size_t allocSize = T.sizeof;
+
+        void* rawMemory = enforce(malloc(allocSize), "malloc returned null");
```

Comment:
@JakobOvrum After a bit of review, I recall that I was following a pattern noticed elsewhere in Phobos:

```
~/s/d/phobos [v2.067.0 ?] % ag 'enforce\(malloc'
std/container/array.d
485:            auto p = enforce(malloc(sz));

std/regex/package.d
565:        _memory = (enforce(malloc(size))[0..size]);
641:            _memory = (enforce(malloc(size))[0..size]);
671:    void[] memory = enforce(malloc(size))[0..size];

std/stdio.d
346:        _p = cast(Impl*) enforce(malloc(Impl.sizeof), "Out of memory");

std/typecons.d
4053:            _store = cast(Impl*) enforce(malloc(Impl.sizeof));

std/uni.d
1745:        auto ptr = cast(T*)enforce(malloc(T.sizeof*size), "out of memory on C heap");
6875:            auto p = cast(ubyte*)enforce(malloc(raw_cap));
6917:        ubyte* p = cast(ubyte*)enforce(malloc(3*(grow+1)));
```

Should these be corrected in another PR?